pkg/cmd/deadman/operator_staleness_check.go

```diff
@@ -0,0 +1,177 @@
+
+	durationAllowedForOperatorResponse time.Duration
+
+	clusterOperatorClient configv1versionedtypes.ClusterOperatorInterface
+	clusterOperatorLister configv1lister.ClusterOperatorLister
+
+	queue workqueue.RateLimitingInterface
+}
+
+// NewOperatorStalenessChecker is a controller that sets operator conditions status, reason, and messages when the operator
+// fails to respond to a challenge and correct a message.
+func NewOperatorStalenessChecker(
+	configClient configeversionedclient.Interface,
+	configInformers configv1informers.SharedInformerFactory,
+	durationAllowedForOperatorResponse time.Duration,
+) *OperatorStalenessChecker {
+
+	c := &OperatorStalenessChecker{
+		operatorToDeadlineForResponse:      map[string]time.Time{},
+		durationAllowedForOperatorResponse: durationAllowedForOperatorResponse,
+
+		clusterOperatorClient: configClient.ConfigV1().ClusterOperators(),
+		clusterOperatorLister: configInformers.Config().V1().ClusterOperators().Lister(),
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "OperatorStalenessChecker"),
+	}
+
+	configInformers.Config().V1().ClusterOperators().Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				clusterOperator := obj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				clusterOperator := newObj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+		},
+		1*time.Minute,
+	)
+
+	return c
+}
+
+func isCheckingForStaleness(clusterOperator *configv1.ClusterOperator) bool {
+	for _, condition := range clusterOperator.Status.Conditions {
+		if strings.HasPrefix(condition.Message, "Checking for stale status") {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (c *OperatorStalenessChecker) syncHandler(ctx context.Context, key string) error {
+	clusterOperator, err := c.clusterOperatorLister.Get(key)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if !isCheckingForStaleness(clusterOperator) {
+		delete(c.operatorToDeadlineForResponse, clusterOperator.Name)
+		return nil
+	}
+
+	now := time.Now()
+	if _, ok := c.operatorToDeadlineForResponse[clusterOperator.Name]; !ok {
+		c.operatorToDeadlineForResponse[clusterOperator.Name] = now.Add(c.durationAllowedForOperatorResponse)
+	}
+	deadlineForReset := c.operatorToDeadlineForResponse[clusterOperator.Name]
+	if now.Before(deadlineForReset) {
+		return nil
+	}
+
+	// if we're past the deadline and some messsages still indicate we're checking for staleness, every condition should
+	// be marked as unknown and indicated as stale.
+	clusterOperatorToWriteAsStale := clusterOperator.DeepCopy()
+	for i, condition := range clusterOperator.Status.Conditions {
+		switch condition.Type {
+		case configv1.OperatorDegraded:
+			// if the operator is not setting status, it is Degraded.
```

Comment:
One can argue that the "degraded" state indicates the "operand" is malfunctioning, not the operator. But I think it is fine, otherwise we are going to invent meta-operators here :-)